packages/cli/src/lib/autorun/autorun-command.unit.test.ts

```diff
@@ -0,0 +1,117 @@
+import { bundleRequire } from 'bundle-require';
+import { vol } from 'memfs';
+import { beforeEach, describe, expect, it, vi } from 'vitest';
+import {
+  PortalUploadArgs,
+  ReportFragment,
+  uploadToPortal,
+} from '@code-pushup/portal-client';
+import { collectAndPersistReports } from '@code-pushup/core';
+import { report } from '@code-pushup/models/testing';
+import { DEFAULT_CLI_CONFIGURATION } from '../../../test/constants';
+import { yargsCli } from '../yargs-cli';
+import { yargsAutorunCommandObject } from './autorun-command';
+
+// This in needed to mock the API client used inside the upload function
+vi.mock('@code-pushup/portal-client', async () => {
+  const module: typeof import('@code-pushup/portal-client') =
+    await vi.importActual('@code-pushup/portal-client');
+
+  return {
+    ...module,
+    uploadToPortal: vi.fn(
+      async () => ({ packageName: '@code-pushup/cli' } as ReportFragment),
+    ),
+  };
+});
+
+// Mock file system API's
+vi.mock('fs', async () => {
+  const memfs: typeof import('memfs') = await vi.importActual('memfs');
+  return memfs.fs;
+});
+vi.mock('fs/promises', async () => {
+  const memfs: typeof import('memfs') = await vi.importActual('memfs');
+  return memfs.fs.promises;
+});
+
+vi.mock('@code-pushup/core', async () => {
+  const core = await vi.importActual('@code-pushup/core');
+  return {
+    ...(core as object),
```

Comment:
Tiny thing, but we can easily make the type more specific.

```suggestion
  const core: typeof import('@code-pushup/core') =
    await vi.importActual('@code-pushup/core');
  return {
    ...core,
```

Reply:
When I tried to do that, I got the `Static imports of lazy-loaded libraries are forbidden` lint error :(

Reply:
Oh, interesting :D OK, then leave it as is, it doesn't matter very much in this case.